src/backend/common/RenderPass.h

```diff
@@ -0,0 +1,72 @@
+// Copyright 2017 The NXT Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BACKEND_COMMON_RENDERPASS_H_
+#define BACKEND_COMMON_RENDERPASS_H_
+
+#include "Forward.h"
+#include "RefCounted.h"
+
+#include "nxt/nxtcpp.h"
+
+#include <array>
+#include <bitset>
+#include <map>
+#include <vector>
+
+namespace backend {
+
+    class RenderPassBase : public RefCounted {
+        public:
+            RenderPassBase(RenderPassBuilder* builder);
+
+            struct SubpassInfo {
+                std::map<uint32_t, uint32_t> colorAttachments;
```

Comment:
std::array<kMaxColorAttachments, uint32_t>
will need to update the code in CommandBuffer that iterates over it to check for nulltpr